File: test/fake_telegram_repository_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:tgthird/src/data/fake_telegram_repository.dart';
import 'package:tgthird/src/models/telegram_models.dart';

void main() {
  test('login moves from code challenge to ready', () async {
    final repository = FakeTelegramRepository();

    await repository.startLogin(
      const LoginCredentials(
        apiId: 1,
        apiHash: 'hash',
        phoneNumber: '+85200000000',
      ),
    );
    expect(repository.authState.phase, AuthPhase.waitingCode);

    await repository.submitCode('12345');
    expect(repository.authState.phase, AuthPhase.ready);
  });

  test('sendText appends a local outgoing message', () async {
    final repository = FakeTelegramRepository();
    final before = repository.messagesFor('saved').length;

    await repository.sendText(chatId: 'saved', text: 'hello');

    final messages = repository.messagesFor('saved');
    expect(messages.length, before + 1);
    expect(messages.last.text, 'hello');
    expect(messages.last.outgoing, isTrue);
  });
}

File: core/src/test/kotlin/com/sing/tgthird/core/repository/FakeTelegramRepositoryTest.kt
package com.sing.tgthird.core.repository

import com.sing.tgthird.core.model.AuthState
import com.sing.tgthird.core.model.LoginCredentials
import kotlinx.coroutines.flow.first
import kotlinx.coroutines.test.runTest
import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test

class FakeTelegramRepositoryTest {
    @Test
    fun loginMovesFromCodeChallengeToReady() = runTest {
        val repository = FakeTelegramRepository()

        repository.startLogin(LoginCredentials(apiId = 1, apiHash = "hash", phoneNumber = "+85200000000"))
        assertEquals(AuthState.WaitingForCode("+85200000000"), repository.authState.value)

        repository.submitCode("12345")
        assertTrue(repository.authState.value is AuthState.Ready)
    }

    @Test
    fun sendTextAppendsMessage() = runTest {
        val repository = FakeTelegramRepository()
        val before = repository.messagesFor("saved").first().size

        repository.sendText(chatId = "saved", text = "hello")

        val messages = repository.messagesFor("saved").first()
        assertEquals(before + 1, messages.size)
        assertEquals("hello", messages.last().text)
    }
}
